SCClassLibrary/Common/Quarks/Quarks.sc

```diff
@@ -203,22 +204,31 @@ Quarks {
 		});
 
 		"Installing %".format(quark.name).postln;
-
+		// add currently installing quark to 'installing' Array
+		installing = (installing.size == 0).if{ quark.bubble }{ installing ++ quark.bubble };
 		quark.checkout();
-		if(quark.isCompatible().not, {
+		quark.isCompatible().not.if{
 			^incompatible.value(quark.name);
-		});
-		quark.dependencies.do { |dep|
-			var ok = dep.install();
-			if(ok.not, {
-				("Failed to install" + quark.name).error;
-				^false
-			});
+		};
+		quark.dependencies.do{ |dep|
+			var ok;
+			// check if dependency is already installing
+			installing.detect({ |q| q.name == dep.name }).isNil.if{				
+				ok = dep.install();
+				ok.not.if{
+					("Failed to install" + quark.name).error;
+					^false
+				}
+			}{
+				(dep.name + "already installing").postln
```

Comment:
Is this posted message useful in practice, or was it for debugging?

Reply:
no you're right - I'll remove